src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -264,36 +269,55 @@ def _backpressure(
             n_finished += len(finished_jobs)
         return n_finished
 
+    @singledispatchmethod
+    @staticmethod
+    def _chunkify(data: Any, n_chunks: int):
+        raise NotImplementedError(
+            f"_chunkify does not support data of type {type(data)}"
+        )
```

Comment:
Much as I like to see overloading instead of ugly elif switches, wouldn't this cause a problem with inheritance? If we wanted to add a new MapReduceJob which chunkifies differently, it's unclear where to register the new method and how it affects other inheriting classes, see [this discussion](https://stackoverflow.com/questions/68190543/how-to-use-singledispatchmethod-with-inheritance-class). OTOH, this is perhaps not that much of an issue, at least right now

Reply:
That's something that we don't need right now and we if we do eventually need the suggestions in that discussion indicate that it is indeed possible without too much effort.

Reply:
But the "solutions" in that thread are not acceptable IMO. If we leave the singledispatch then we are consciously disallowing extension of MapReduce by users.

Reply:
I don't agree with allowing users to extend MapReduceJob since it is meant to be transparent to the users. Still I will remove it.